sql/src/main/java/org/apache/druid/sql/calcite/planner/DruidPlanner.java

```diff
@@ -645,11 +661,14 @@ private RelNode rewriteRelDynamicParameters(RelNode rootRel)
 
   private QueryMaker buildQueryMaker(
       final RelRoot rootQueryRel,
-      @Nullable final SqlInsert insert
+      @Nullable final SqlInsert insert,
```

Comment:
can this take `insertOrReplace` `SqlInsert` node? and then we can assign `DruidSqlInsert` or `DruidSqlReplace` depending on the instance check as is done now.

Reply:
This is a better design. I have change it so that it is only checked during the creation of ParsedNode and to add it to the context.